packages/react-navigation/lib/complete-navigation.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { PropsWithChildren } from 'react';
+import { NavigationContext } from './navigation-context';
+
+interface Props extends PropsWithChildren {
+  on: 'mount' | 'unmount';
+}
+
+/** End the current navigation span when the component is mounted, unmounted or the `on` prop is `true` */
+export class CompleteNavigation extends React.Component<Props> {
+  static contextType = NavigationContext;
+  
+  // @ts-expect-error
+  context: React.ContextType<typeof NavigationContext>;
+
+  constructor(props: Props, context: typeof NavigationContext) {
+    super(props);
+
+    // @ts-expect-error
+    context.blockNavigationEnd();
+  }
```

Comment:
I think these typescript errors can be fixed by the following?
https://legacy.reactjs.org/docs/context.html#classcontexttype

```suggestion
export class CompleteNavigation extends React.Component<Props> {
  static contextType = NavigationContext;
  
  declare context: React.ContextType<typeof NavigationContext>;

  constructor(props: Props) {
    super(props);

    this.context.blockNavigationEnd();
  }
```

Reply:
ah maybe not, that breaks the unit tests...

Reply:
updated!